### DISCUSSION TITLE
Wrapper hooks have different states for same key

### DISCUSSION THREAD

**Discussion by ajones55555** [vercel/swr#1026] [Q&A] [answered]
In this example with wrapper hooks, why doesn't the upper text update when the bottom button is pressed?

https://codesandbox.io/s/useswr-no-rerender-ml17e?file=/src/App.js

Sorry for the contrived example, working in a big codebase.

Thanks in advance for any help!

**Comment by ErnestBrandi** [comment]
Hi, I guess it's because when you declare `variables`, even if use useMemo, the created object is not the same in the two `useBase` calls so it results as being different keys from SWR point of view (check out the very bottom lines of this page : https://swr.vercel.app/docs/arguments#passing-objects). Try passing it down as a prop (from App for instance).

**Reply by ajones55555** [reply]
yeah, you're right. This is the solution we are going with: https://github.com/vercel/swr/discussions/1026#discussioncomment-474359

**Comment by ajones55555** [comment, marked as answer]
For now we're solving this with a system like this one as a workaround.
https://github.com/vercel/commerce/blob/master/framework/commerce/utils/use-data.tsx
